Tasks/VSBuild/VSBuild.ps1

```diff
@@ -1,207 +1,74 @@
+[cmdletbinding()]
 param(
-    [string]$vsLocation, # Support for vsLocation has been deprecated.
-    [string]$vsVersion,
-    [string]$msBuildLocation, # Support for msBuildLocation has been deprecated.
-    [string]$msBuildVersion, # Support for msBuildVersion has been deprecated.
-    [string]$msBuildArchitecture,
-    [string]$msBuildArgs,
-    [string]$solution, 
-    [string]$platform,
-    [string]$configuration,
-    [string]$clean,
-    [string]$restoreNugetPackages,
-    [string]$logProjectEvents
-)
+    [string]$VSLocation, # Support for vsLocation has been deprecated.
+    [string]$VSVersion,
+    [string]$MSBuildLocation, # Support for msBuildLocation has been deprecated.
+    [string]$MSBuildVersion, # Support for msBuildVersion has been deprecated.
+    [string]$MSBuildArchitecture,
+    [string]$MSBuildArgs,
+    [string]$Solution,
+    [string]$Platform,
+    [string]$Configuration,
+    [string]$Clean,
+    [string]$RestoreNugetPackages,
+    [string]$LogProjectEvents,
+    [string]$OmitDotSource)
 
 Write-Verbose "Entering script VSBuild.ps1"
-Write-Verbose "vsLocation = $vsLocation"
-Write-Verbose "vsVersion = $vsVersion"
-Write-Verbose "msBuildLocation = $msBuildLocation"
-Write-Verbose "msBuildVersion = $msBuildVersion"
-Write-Verbose "msBuildArchitecture = $msBuildArchitecture"
-Write-Verbose "msBuildArgs = $msBuildArgs"
-Write-Verbose "solution = $solution"
-Write-Verbose "platform = $platform"
-Write-Verbose "configuration = $configuration"
-Write-Verbose "clean = $clean"
-Write-Verbose "restoreNugetPackages = $restoreNugetPackages"
-Write-Verbose "logProjectEvents = $logProjectEvents"
+Write-Verbose "VSLocation = $VSLocation"
+Write-Verbose "VSVersion = $VSVersion"
+Write-Verbose "MSBuildLocation = $MSBuildLocation"
+Write-Verbose "MSBuildVersion = $MSBuildVersion"
+Write-Verbose "MSBuildArchitecture = $MSBuildArchitecture"
+Write-Verbose "MSBuildArgs = $MSBuildArgs"
+Write-Verbose "Solution = $Solution"
+Write-Verbose "Platform = $Platform"
+Write-Verbose "Configuration = $Configuration"
+Write-Verbose "Clean = $Clean"
+Write-Verbose "RestoreNugetPackages = $RestoreNugetPackages"
+Write-Verbose "LogProjectEvents = $LogProjectEvents"
 
 # Import the Task.Common and Task.Internal dll that has all the cmdlets we need for Build
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 
-if (!$solution)
-{
-    throw (Get-LocalizedString -Key "Solution parameter not set on script")
-}
-
-$nugetRestore = Convert-String $restoreNugetPackages Boolean
```

Comment:
Since I added unit tests for this logic, I removed all these extra variables that hinder readability. Since the variables are originally strongly typed as strings, I re-type them as bool when the new value is assigned.